src/lib/FreeC/Environment.hs

```diff
@@ -188,6 +194,16 @@ usedIdents = concatMap entryIdents . Map.elems . envEntries
     | isConEntry entry = [entryIdent entry, entrySmartIdent entry]
     | otherwise        = [entryIdent entry]
 
+-- | Gets a list of Agda identifiers for functions, (type/smart) constructors,
+--   (type/fresh) variables that were used in the given environment already.
+usedAgdaIdents :: Environment -> [Agda.QName]
+usedAgdaIdents = concatMap entryIdents . Map.elems . envEntries
+ where
+  entryIdents :: EnvEntry -> [Agda.QName]
+  entryIdents entry
+    | isConEntry entry = [entryAgdaIdent entry, entryAgdaSmartIdent entry]
+    | otherwise        = [entryAgdaIdent entry]
```

Comment:
Again, not sure it adds readability, but perhaps it's more explicit about the parts all cases have in common.

```suggestion
  entryIdents entry = (if isConEntry entry then (entryAgdaSmartIdent entry :) else id) [entryAgdaIdent entry]
```

Reply:
In this case we could use `guard` to avoid the `else id`.
```haskell
  entryIdents entry =
    [entryAgdaIdent entry]
      ++ (guard (isConEntry entry) $> entryAgdaSmartIdent entry)
```